src/asInput/index.jsx

```diff
@@ -204,6 +209,30 @@ const asInput = (WrappedComponent, inputType = undefined, labelFirst = true) =>
       );
     }
 
+    renderInputGroupAddons(type) {
+      // Type is either inputGroupPrepend or inputGroupAppend
+      const formattedType = type.replace('inputGroup', '').toLowerCase();
+      const addons = this.props[type];
+      if (Array.isArray(addons)) {
+        const addonsChildren = addons.map((addon, index) => (
+          React.cloneElement(
+            addon,
+            { key: this.generateInputGroupAddonKey({ prefix: formattedType, index }) },
+          )
+        ));
+        return (
+          <div className={styles[`input-group-${formattedType}`]}>
```

Comment:
If you create an `getAddons` method that deals with the `array` / not array cases then you can get rid of some duplicative code by doing

```javascript
return (
  <div className={style[`input-group-${formattedType}`]}
    { this.getAddons() }
  </div>
)
```